src/erlcloud_ddb_util.erl

```diff
@@ -477,6 +501,52 @@ batch_write_retry(RequestItems, Config) ->
             batch_write_retry(Unprocessed, Config)
     end.
 
+%%------------------------------------------------------------------------------
+%% @doc
+%%  wait until table_status==active.
+%%  RetryTimes = 0 means infinity.
+%%
+%% ===Example===
+%%
+%% `
+%% erlcloud_ddb2:wait_for_table_active(<<"TableName">>, 3000, 40, Config)
+%% '
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec wait_for_table_active(table_name(), pos_integer(), non_neg_integer(), aws_config()) ->
+    ok | {error, deleting | timeout | any()}.
+wait_for_table_active(Table, Interval, RetryTimes, Config) when is_binary(Table), Interval > 0, RetryTimes >= 0 ->
+    case erlcloud_ddb2:describe_table(Table, [{out, record}], Config) of
+        {ok, #ddb2_describe_table{table = #ddb2_table_description{table_status = active}}} ->
+            ok;
+        {ok, #ddb2_describe_table{table = #ddb2_table_description{table_status = deleting}}} ->
+            {error, deleting};
+        {ok, _} ->
+            case RetryTimes of
+                0 ->
+                    timer:sleep(Interval),
+                    wait_for_table_active(Table, Interval, RetryTimes, Config);
+                1 ->
+                    {error, timeout};
+                _ ->
+                    timer:sleep(Interval),
+                    wait_for_table_active(Table, Interval, RetryTimes - 1, Config)
+            end;
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+wait_for_table_active(Table, Interval, RetryTimes) ->
+    wait_for_table_active(Table, Interval, RetryTimes, default_config()).
+
+wait_for_table_active(Table, AWSCfg) ->
+    wait_for_table_active(Table, 3000, 0, AWSCfg).
```

Comment:
IMO infinity is not a good default here.

Reply:
I don't think so but I accept your suggestion.